components/validator/resources.rst

```diff
@@ -0,0 +1,200 @@
+        ->addMethodMapping('loadValidatorMetadata')
+        ->getValidator();
+
+Now, the retrieved ``Validator`` tries to find the ``loadValidatorMetadata()``
+method of the class to validate to load its metadata::
+
+    use Symfony\Component\Validator\Mapping\ClassMetadata;
+    use Symfony\Component\Validator\Constraints\NotBlank;
+    use Symfony\Component\Validator\Constraints\Length;
+
+    class User
+    {
+        protected $name;
+
+        public static function loadValidatorMatadata(ClassMetadata $metadata)
+        {
+            $metadata->addPropertyConstraint('name', new NotBlank());
+            $metadata->addPropertyConstraint('name', new Length(array(
+                'min' => 5,
+                'max' => 20,
+            )));
+        }
+    }
+
+.. tip::
+
+    You can call this method multiple times to add multiple supported method
+    names. You can also use
+    :method:`Symfony\\Component\\Validator\\ValidatorBuilder::addMethodMappings`
+    to set an array of supported method names.
+
+The FileLoaders
+---------------
+
+The component also provides 2 file loaders, one to load Yaml files and one to
+load XML files. Use 
+:method:`Symfony\\Component\\Validator\\ValidatorBuilder::addYamlMapping` or
+:method:`Symfony\\Component\\Validator\\ValidatorBuilder::addXmlMapping` to
+configure the locations of these files::
+
+    use Symfony\Component\Validator\Validation;
+
+    $validator = Validation::createValidatorBuilder()
+        ->addYamlMapping('config/validation.yml')
+        ->getValidator();
+
+.. tip::
+
+    Just like with the method mappings, you can also use 
+    :method:`Symfony\\Component\\Validator\\ValidatorBuilder::addYamlMappings` and
+    :method:`Symfony\\Component\\Validator\\ValidatorBuilder::addXmlMappings`
+    to configure an array of file paths.
+
+The AnnotationLoader
+--------------------
+
+At last, the component provides an
+:class:`Symfony\\Component\\Validator\\Mapping\\Loader\\AnnotationLoader`.
+This loader uses an annotation reader to parse the annotations of a class.
+Annotations are placed in doc block comments (`/** ... */`) and start with an
+``@``. For instance::
+
+    // ...
+
+    /**
+     * @Assert\NotBlank()
+     */
+    protected $name;
+
+To enable the annotation loader, call the 
+:method:`Symfony\\Component\\Validator\\ValidatorBuilder::enableAnnotationMapping`
+method. It takes an optional annotation reader instance, which defaults to
+``Doctrine\Common\Annotations\AnnotationReader``::
+
+    use Symfony\Component\Validator\Validation;
+
+    $validator = Validation::createValidatorBuilder()
+        ->enableAnnotationMapping()
+        ->getValidator();
+
+To disable the annotation loader after it was enabled, call
+:method:`Symfony\\Component\\Validator\\ValidatorBuilder::disableAnnotationMapping`.
+
+.. note::
+
+    In order to use the annotation loader, you should have installed the
+    ``doctrine/annotations`` and ``doctrine/cache`` packages from Packagist.
```

Comment:
maybe add the require command here as well: `composer require doctrine/cache doctrine/annotations`